Frameworks/CoreText/CTRun.mm

```diff
@@ -215,43 +215,47 @@ CFRange CTRunGetStringRange(CTRunRef run) {
-    CGFloat pointSize = font ? CTFontGetSize(font) : kCTFontSystemFontSize;
-
-    DWRITE_FONT_METRICS fontMetrics;
-    curRun->_dwriteGlyphRun.fontFace->GetMetrics(&fontMetrics);
-
-    // Need scaling factor to convert from design units to pointSize
-    CGFloat scalingFactor = pointSize / fontMetrics.designUnitsPerEm;
-
-    DWRITE_GLYPH_METRICS glyphMetrics[range.length];
-    THROW_IF_FAILED(curRun->_dwriteGlyphRun.fontFace->GetDesignGlyphMetrics(curRun->_dwriteGlyphRun.glyphIndices + range.location,
-                                                                            range.length,
-                                                                            glyphMetrics,
-                                                                            curRun->_dwriteGlyphRun.isSideways));
-
-    double width = 0;
-    CGFloat newAscent = -FLT_MAX;
-    CGFloat newDescent = -FLT_MAX;
-    for (size_t i = range.location; i < range.location + range.length; ++i) {
-        newAscent = std::max(newAscent, glyphMetrics[i].verticalOriginY * scalingFactor);
-        newDescent = std::max(newDescent, glyphMetrics[i].bottomSideBearing * scalingFactor);
-        width += curRun->_dwriteGlyphRun.glyphAdvances[i];
-    }
-
-    if (ascent) {
-        *ascent = newAscent;
-    }
+    if (ascent || descent || leading) {
+        CTFontRef font = static_cast<CTFontRef>([curRun->_attributes objectForKey:static_cast<NSString*>(kCTFontAttributeName)]);
+        CGFloat pointSize = font ? CTFontGetSize(font) : kCTFontSystemFontSize;
+
+        DWRITE_FONT_METRICS fontMetrics;
+        curRun->_dwriteGlyphRun.fontFace->GetMetrics(&fontMetrics);
+
+        // Need scaling factor to convert from design units to pointSize
+        CGFloat scalingFactor = pointSize / fontMetrics.designUnitsPerEm;
+
+        DWRITE_GLYPH_METRICS glyphMetrics[range.length];
+        THROW_IF_FAILED(curRun->_dwriteGlyphRun.fontFace->GetDesignGlyphMetrics(curRun->_dwriteGlyphRun.glyphIndices + range.location -
+                                                                                    curRun->_range.location,
+                                                                                range.length,
+                                                                                glyphMetrics,
+                                                                                curRun->_dwriteGlyphRun.isSideways));
+
+        double width = 0;
+        CGFloat newAscent = -FLT_MAX;
+        CGFloat newDescent = FLT_MAX;
+        for (size_t i = range.location - curRun->_range.location; i < range.location + range.length - curRun->_range.location; ++i) {
+            newAscent = std::max(newAscent, glyphMetrics[i].verticalOriginY * scalingFactor);
+            newDescent = std::min(newDescent, -glyphMetrics[i].bottomSideBearing * scalingFactor);
+            width += curRun->_dwriteGlyphRun.glyphAdvances[i];
```

Comment:
looks like some I missed some other changes that got reverted when git blew up on me